cnf-certification-test/networking/suite.go

```diff
@@ -124,6 +132,33 @@ func testListenAndDeclared(env *provider.TestEnvironment) {
 	}
 }
 
+func testNodePort(env *provider.TestEnvironment) {
+	badNamespaces := []string{}
+	badServices := []string{}
+	client := clientsholder.GetClientsHolder()
+	for _, ns := range env.Namespaces {
+		ginkgo.By(fmt.Sprintf("Testing services in namespace %s", ns))
+		services, err := client.K8sClient.CoreV1().Services(ns).List(context.TODO(), metav1.ListOptions{})
+		if err != nil {
+			tnf.ClaimFilePrintf("Failed to list services on namespace %s, Error: %v", ns, err)
+			badNamespaces = append(badNamespaces, ns)
+			continue
+		}
+		for i := range services.Items {
+			service := &services.Items[i]
+			if service.Spec.Type == nodePort {
+				serviceName := service.ObjectMeta.Name
+				serviceNamespace := service.ObjectMeta.Namespace
+				tnf.ClaimFilePrintf("FAILURE: Service %s (ns %s) type is nodePort", serviceName, serviceNamespace)
+				badServices = append(badServices, fmt.Sprintf("ns: %s, name: %s", serviceNamespace, serviceName))
+			}
+		}
+	}
+	if ns, bs := len(badNamespaces), len(badServices); ns > 0 || bs > 0 {
+		ginkgo.Fail(fmt.Sprintf("%d namespaces have nodePort and %d services type is nodePort", ns, bs))
```

Comment:
Since the badNamespaces is only used to save them when the List command failed, I suggest changing this trace to:
```
"Failed to get services on %d namespaces. %d services found of type nodePort."
```